jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainFragment.java

```diff
@@ -58,22 +61,28 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         editMessage = (EditText) getView().findViewById(R.id.edit_text_message);
         showHideButton = (Button) getView().findViewById(R.id.button_showhide);
         sendButton = (Button) getView().findViewById(R.id.button_send);
+        showAllMessagesButton = (Button) getView().findViewById(R.id.button_show_all_messages);
 
         // Set listeners
         showHideButton.setOnClickListener(this);
         sendButton.setOnClickListener(this);
+        showAllMessagesButton.setOnClickListener(this);
```

Comment:
This approach is often preferred as it avoids the stuff you have now in `onClick` which becomes more and more cluttered as we add more listeners